### PR TITLE
refactor: Refactor scan.py to remove code duplicates

### PR DESCRIPTION
**Pull request by baturayo**
Resolves: https://sodadata.atlassian.net/browse/CLOUD-216

### CODE REVIEW

File: soda/core/soda/execution/data_source_scan.py [hunk 1]
```diff
@@ -65,16 +71,24 @@ def execute_queries(self):
...
+            profile_columns_run = ProfileColumnsRun(self, data_source_check_cfg).run()
+            scan._profile_columns_result_tables.extend(profile_columns_run.tables)
+            scan._is_profiling_run = True  # TODO: delete this line
 
-    def create_discover_tables_run(self, data_source_check_cfg, scan):
-        return DiscoverTablesRun(self, data_source_check_cfg)
+        if isinstance(data_source_check_cfg, DiscoverTablesCfg):
+            discover_tables_run = DiscoverTablesRun(self, data_source_check_cfg).run()
+            scan._discover_tables_result_tables.extend(discover_tables_run.tables)
+            scan._is_profiling_run = True  # TODO: delete this line
```
Comment (bastienboutonnet) [inline soda/core/soda/execution/data_source_scan.py:89]: What happens when you remove this?

**Review by bastienboutonnet — COMMENTED** [review]
Really great cleanup! One question about the is profiling issue, I want to make sure I understand why you have a todo that says remove it but it's still there.

File: soda/core/soda/execution/data_source_scan.py (hunk 1 above)
Comment (baturayo) [reply soda/core/soda/execution/data_source_scan.py:89]: the unit tests will fail in the current code. If we merge your [PR](https://github.com/sodadata/soda-core/pull/1377), then I'll delete this line

Comment (bastienboutonnet) [reply soda/core/soda/execution/data_source_scan.py:89]: What in my PR helps? The test assertion helper?

File: soda/core/tests/data_source/test_anomaly_check.py [hunk 2]
```diff
@@ -17,7 +20,7 @@ def test_anomaly_detection_default(scanner: Scanner):
 
     scan.mock_historic_values(
         metric_identity=f"metric-{scan._scan_definition_name}-{scan._data_source_name}-{table_name}-row_count",
-        metric_values=[10, 10, 10, 9, 8, 8, 8, 0, 0, 0],
+        metric_values=[10, 10, 10, 9, 8, 8, 8, 10, 10, 10],
```
[the remark this answers is not in the document]
Comment (baturayo) [reply soda/core/tests/data_source/test_anomaly_check.py:23]: ah I was trying smth. Good that you noticed, I ll revert

**Review by bastienboutonnet — APPROVED** [review]
LGTM

Here's a "shipit cat" (it's a cat most of the time)

**Review by bastienboutonnet — APPROVED** [review]
Ok one more approval from me. Please check with @vijaykiran with merging.

Here's a "shipit cat" (it's a cat most of the time)

**Pull request merged**